libcaf_net/test/net/web_socket/framing.cpp

```diff
@@ -0,0 +1,184 @@
+// This file is part of CAF, the C++ Actor Framework. See the file LICENSE in
+// the main distribution directory for license terms and copyright or visit
+// https://github.com/actor-framework/actor-framework/blob/master/LICENSE.
+
+#define CAF_SUITE net.web_socket.framing
+
+#include "caf/net/web_socket/framing.hpp"
+
+#include "net-test.hpp"
+
+using namespace caf;
+using namespace std::literals;
+
+namespace {
+
+class app_t : public net::web_socket::upper_layer {
+public:
+  static auto make() {
+    return std::make_unique<app_t>();
+  }
+
+  std::string text_input;
+
+  caf::byte_buffer binary_input;
+
+  error err;
+
+  bool aborted = false;
+
+  net::web_socket::lower_layer* down;
+
+  error start(net::web_socket::lower_layer* ll) override {
+    down = ll;
+    return none;
+  }
+
+  void prepare_send() override {
+    // nop
+  }
+
+  bool done_sending() override {
+    return true;
+  }
+
+  void abort(const error& reason) override {
+    aborted = true;
+    err = reason;
+    down->shutdown(reason);
+  }
+
+  ptrdiff_t consume_text(std::string_view text) override {
+    text_input.insert(text_input.end(), text.begin(), text.end());
+    return static_cast<ptrdiff_t>(text.size());
+  }
+
+  ptrdiff_t consume_binary(byte_span bytes) override {
+    binary_input.insert(binary_input.end(), bytes.begin(), bytes.end());
+    return static_cast<ptrdiff_t>(bytes.size());
+  }
+};
+
+struct fixture {
+  app_t* app;
+  net::web_socket::framing* uut;
+  std::unique_ptr<mock_stream_transport> transport;
+
+  fixture() {
+    auto app_layer = app_t::make();
+    app = app_layer.get();
+    auto uut_layer
+      = net::web_socket::framing::make_server(std::move(app_layer));
+    uut = uut_layer.get();
+    transport = mock_stream_transport::make(std::move(uut_layer));
+  }
+
+  static const_byte_span make_test_data(uint64_t requested_size) {
+    static std::vector<std::byte> bytes{150, std::byte{0xFF}};
+    if (requested_size > bytes.size())
+      bytes.resize(requested_size, std::byte{0xFF});
+    return const_byte_span{bytes.data(), requested_size};
+  }
```

Comment:
I don't know if static test data is acceptable, maybe I could elevate this into net-test and use it in rfc6455 tests? Or just to implement it in that file as well?